library/query_unpinnable_action.yml

```diff
@@ -0,0 +1,25 @@
+id: RP-12
+
+info:
+  name: Unpinnable Action
+  severity: medium
+  description: Unpinnable actions can lead to software supply chain attacks.
+  full-description: |
+    Actions can be pinned to a specific version to ensure that the same version is used
+    every time the workflow is run. Even if the action is pinned, if that action itself
+    uses an unpinned Docker Image or GitHub Action, the action can be updated without
+    the workflow being updated. This can lead to software supply chain attacks.
+  references:
+    - https://www.paloaltonetworks.com/blog/prisma-cloud/unpinnable-actions-github-security/
+  tags:
+    - supply-chain
```

Comment:
We Can do something like: 
``` cypher
MATCH (ca:CompositeAction)
  WHERE (
      ca.using = "docker" AND (
        NOT ca.image CONTAINS "@sha256:"
      )
  )
  RETURN DISTINCT ca;
```

It will return many results - meaning all actions that are using actions that do not pin to specific sha.

Reply:
Due to many results -> downgrade from `medium` to `low`